src/Tool-SystemReporter/SizeReporter.class.st

```diff
@@ -0,0 +1,84 @@
+"
+A subclass of AbstractReporter that computes various metrics over the image
+"
+Class {
+	#name : #SizeReporter,
+	#superclass : #AbstractReporter,
+	#category : #'Tool-SystemReporter'
+}
+
+{ #category : #'instance creation' }
+SizeReporter class >> open [
+	<script>
+	^ super open
+]
+
+{ #category : #'system menu' }
+SizeReporter class >> systemReporterMenuOn: aBuilder [ 
+	<worldMenu>
+	
+	(aBuilder item: #'Size Reporter')
+		parent: #System;
+		action: [ self open ];
+		help: 'Reports different numbers on the size of the image.'
+]
+
+{ #category : #initialization }
+SizeReporter >> initialize [
+	super initialize.
+	self
+		add: #'# classes' method: #reportNbClasses:;
+		add: #'# traits' method: #reportNbTraits:;
+		add: #'Total LOC' method: #reportLOC:.
+	categoriesSelected := Set with: #'# classes'.
+	self updateReport
+]
+
+{ #category : #accessing }
+SizeReporter >> label [
+	^ 'Size Reporter' 
+]
+
+{ #category : #'instance creation' }
+SizeReporter >> open [
+	<script: 'self new open'>
+	super open
+]
+
+{ #category : #reporting }
+SizeReporter >> reportLOC: aStream [
+	self header: 'Number of LinesOfCode' on: aStream.
+	aStream 
+		nextPutAll: 'a lot !' ; cr
+
+]
+
+{ #category : #reporting }
+SizeReporter >> reportNbClasses: aStream [
+	| imageClasses |
+	imageClasses := Smalltalk globals allClasses.
```

Comment:
you should introduce #environment variable in superclass and use it here:
```Smalltalk
AbstractReporter>>initialize
   ...
   environment := self class environment

SizeReporter>>reportNbClasses: aStream
    classes := environment allClasses
```

Reply:
Also please use full names. "Nb" here is not good.

Reply:
agreed